rotkehlchen/db/updates.py

```diff
@@ -88,49 +90,111 @@ def update_spam_assets(self, local_version: int, to_version: int) -> None:
-            file_url = SPAM_ASSETS_URL.format(branch=self.branch, version=version)
-            try:
-                updates = query_file(file_url, is_json=True)
-            except RemoteError as e:
-                log.warning(f'Failed to update spam assets due to {str(e)}')
-                break
-            updated_assets = updates.get('assets')
-            if updated_assets is None:
-                log.error(f'Remote update {file_url} does not contain the assets key')
+            ok, updated_assets = self._get_file_content(
+                file_url=SPAM_ASSETS_URL.format(branch=self.branch, version=version),
+                update_type=UpdateType.SPAM_ASSETS,
+            )
+            if ok is False:
                 continue
 
-            log.info(f'Applying update for spam assets from {version}. {len(updates)} tokens to add')  # noqa: E501
+            log.info(f'Applying update for spam assets from v{version}. {len(updated_assets)} tokens to add')  # noqa: E501
             with GlobalDBHandler().conn.critical_section():
                 # Use a critical section to avoid another greenlet adding spam assets at
                 # the same time
-                update_spam_assets(db=self.user_db, assets_info=updates['assets'])
+                update_spam_assets(db=self.user_db, assets_info=updated_assets)
 
             with self.user_db.conn.write_ctx() as write_cursor:
                 write_cursor.execute(
                     'INSERT OR REPLACE INTO settings(name, value) VALUES (?, ?)',
                     (UpdateType.SPAM_ASSETS.serialize(), version),
                 )
 
+    def update_rpc_nodes(self, local_version: int, to_version: int) -> None:
+        """
+        Check the updates in the inclusive range [local_version + 1, to_version] and update the
+        rpc nodes for those versions.
+        """
+        for version in range(local_version + 1, to_version + 1):
+            ok, updated_rpc_nodes = self._get_file_content(
+                file_url=RPC_NODES_URL.format(branch=self.branch, version=version),
+                update_type=UpdateType.RPC_NODES,
+            )
+            if ok is False:
+                continue
+
+            log.info(f'Applying update for rpc nodes from v{version}')
+            with GlobalDBHandler().conn.write_ctx() as write_cursor:
+                write_cursor.execute('DELETE FROM rpc_nodes')
+                write_cursor.executemany(
+                    'INSERT OR IGNORE INTO rpc_nodes(name, endpoint, owned, active, weight, blockchain) '  # noqa: E501
+                    'VALUES (?, ?, ?, ?, ?, ?)',
+                    [
+                        (node['name'], node['endpoint'], node['owned'], node['active'], str(FVal(node['weight'])), node['blockchain'])  # noqa: E501
+                        for node in updated_rpc_nodes
+                    ],
+                )
+
+            with self.user_db.conn.write_ctx() as write_cursor:
```

Comment:
And as we said in the last PR review, apart from updating the defaults it would make sense to do the following with the user DB:

Check if the endpoint/chain combination exists for the user. If not add it and update the weights.
Removing an old endpoint may be trickier. Not sure if it would make much sense. But perhaps check the previous defaults, compare to the new defaults and see if something is missing. If this missing node is present in the user DB remove it. But not totally sure.

Finally make sure this user rpc nodes editing is in a different function.